aggregator/src/main/java/com/redhat/cloud/notifications/db/EmailAggregationRepository.java

```diff
@@ -43,4 +47,26 @@ public List<AggregationCommand> getApplicationsWithPendingAggregationAccordinfOr
                 .collect(toList());
     }
 
+    public List<IAggregationCommand> getApplicationsWithPendingAggregationAccordingOrgPref(LocalDateTime now) {
+        String query = "SELECT DISTINCT ev.orgId, ev.bundleId, ev.applicationId, acp.lastRun, bu.name, ap.name FROM Event ev " +
+            "join Application ap on ev.applicationId = ap.id join Bundle bu on ev.bundleId = bu.id " +
+            "left join AggregationOrgConfig acp on ev.orgId = acp.orgId " +
+            "WHERE ev.orgId in (SELECT DISTINCT es.id.orgId FROM EventTypeEmailSubscription es where es.id.subscriptionType='DAILY' and es.subscribed = true) " +
+            "AND ev.created > acp.lastRun AND ev.created <= :now " +
+            "AND :nowTime >= acp.scheduledExecutionTime AND (:nowTime - acp.scheduledExecutionTime) < :cutoff";
+        Query hqlQuery = entityManager.createQuery(query)
+            .setParameter("nowTime", now.toLocalTime())
+            .setParameter("cutoff", Duration.ofMinutes(15))
+            .setParameter("now", now);
+
+        List<Object[]> records = hqlQuery.getResultList();
+        return records.stream()
+            .map(emailAggregationRecord -> new EventAggregationCommand(
```

Comment:
Doesn't this work?
```suggestion
            .map(emailAggregationRecord -> new AggregationCommand(
```
If it does, then do we really need `IAggregationCommand` and `EventAggregationCommand`?

Reply:
Yes it could be removed, I will.